### DISCUSSION TITLE
Creating Basic Tests in Neovim with LazyVim for Golang

### DISCUSSION THREAD

**Discussion by skinner12** [LazyVim/LazyVim#937] [Q&A] [answered]
Hello everyone,

I recently switched from Visual Studio Code to Neovim with the LazyVim plugin for my Golang development workflow. I'm loving the new environment and have successfully implemented additional configurations for Go, Neotest, and DAP (Debug Adapter Protocol).

With these extra configurations in place, I'm now able to run tests efficiently and debug my Golang code effectively. However, I'm still struggling with creating basic tests for new files in Golang.

In Visual Studio Code, I could easily generate a test skeleton by clicking on the function I wanted to test. I'm wondering if there's a similar feature or command in LazyVim, along with the added configurations, that allows me to streamline the process of creating basic tests for new Golang files.

I would greatly appreciate it if someone from the community could provide guidance on how to leverage these configurations and tools (Go, Neotest, and DAP) to simplify the creation of basic tests in Golang using Neovim and LazyVim.

Thank you in advance for your help and valuable insights!

Best regards

**Comment by dpetka2001** [comment]
I haven't done this kind of thing myself, but if I would do it (if there is no plugin that offers this) I would do it with Luasnip snippets (assuming that the skeleton template that vscode has is general enough to abstract).

**Comment by binary4cat** [comment, marked as answer]
You need to install the go.nvim plugin, which provides many functions, how to automatically add tests is described in [this part of the document](https://github.com/ray-x/go.nvim#unit-test-with-gotests-and-testify).

**Reply by skinner12** [reply]
Thanks, this response is exactly what I was looking for and what I needed.
